app.py

```diff
@@ -0,0 +1,164 @@
+'''
+Usage:
+    add_person <first_name> <last_name> <role> [--wants_accomodation=N]
+    create_room <room_name> <room_type>
+    reallocate_person <full_name> <new_room_name>
+    load_people <filename>
+    print_people
+    print_allocations [--o=filename]
+    print_unallocated [--o=filename]
+    print_room <room_name>
+    save_state <database>
+    load_state <database>
+    quit
+Options:
+    -h, --help  Show this screen and exit
+    -i --interactive  Interactive Mode
+    --wants_accomodation=<N> [defult: N]
+'''
+
+from app.amity import Amity
+import sys
+import cmd
+import os
+import time
+from termcolor import cprint, colored
+from pyfiglet import figlet_format
+from docopt import docopt, DocoptExit
+
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, arg)
+
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+
+            print('Invalid Command!')
+            print(e)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here.
+
+            return
+
+        return func(self, opt)
+
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+
+border = colored("*" * 20, 'red').center(80)
+
+
+def introduction():
+    print(border)
+    print("AMITY SPACE ALLOCATION!".center(70))
+    print(__doc__)
+    print(border)
+
+
+# def save_state_on_interrupt():
+#     time_stamp = time.strftime("%Y-%m-%d-%H:%M")
+#     db_name = "backup-at-" + time_stamp
+#     print("Saving state to " + db_name)
+#     Amity.save_state(db_name)
+
+
+class AmityApplication(cmd.Cmd):
+    amity = Amity()
+
+    # cprint(figlet_format('AMITY', font='banner3-D'), 'cyan', attrs=['bold'])
+
+    prompt = "(amity)"
+
+    @docopt_cmd
+    def do_create_room(self, arg):
+        '''Usage: create_room <room_name> <room_type>'''
```

Comment:
Nitpick - Use three double quotes. This won't affect how the code runs, it's just a preference of PEP 8.